File: cine/SergioDeIscar-cine-java/SergioDeIscar-cine-java/src/main/java/models/Butaca.java
package models;

import enums.ButacaEstado;
import java.util.Random;

public class Butaca {
    private ButacaEstado estado;
    private boolean isVip = false;

    //region Constructor
    public Butaca(){
        setEstado(ButacaEstado.LIBRE);
        setIsVip(false);
    }
    public Butaca(int probability){
        setEstado(ButacaEstado.LIBRE);
        Random rdn = new Random();
        setIsVip(rdn.nextInt(100) < probability);
    }
    //endregion

    //region Getter
    public ButacaEstado getEstado(){
        return estado;
    }
    public boolean getIsVip(){
        return isVip;
    }
    //endregion

    //region Setter
    public void setEstado(ButacaEstado newEstado){
        estado = newEstado;
    }
    public void setIsVip(boolean newValue){
        isVip = newValue;
    }
    //endregion
}

File: cine/SergioDeIscar-cine-kotlin/SergioDeIscar-cine-kotlin/src/main/kotlin/salacine/models/Butaca.kt
package salacine.models

import salacine.enums.ButacaEstado

class Butaca(private var estado: ButacaEstado){
    private var isVip: Boolean = false

    constructor(estado: ButacaEstado, probability: Int):  this(estado){
        this.isVip = (0..100).random() in 0..probability
    }

    fun getEstado(): ButacaEstado {
        return estado
    }
    fun setEstado(newEstado: ButacaEstado){
        estado = newEstado
    }
    fun getIsVip(): Boolean {
        return isVip
    }
}
